python/dgl/data/movielens.py

```diff
@@ -0,0 +1,678 @@
+                    int(self.test_graph.num_edges() / 2)
+                )
+            )
+
+    def __getitem__(self, idx):
+        assert (
+            idx == 0
+        ), "This dataset has only one set of training, validation and testing graph"
+        if self._transform is None:
+            return self.train_graph, self.valid_graph, self.test_graph
+        else:
+            return (
+                self._transform(self.train_graph),
+                self._transform(self.valid_graph),
+                self._transform(self.test_graph),
+            )
+
+    def __len__(self):
+        return 1
+
+    @property
+    def raw_path(self):
+        if self.name == "ml-10m":
+            return os.path.join(self.raw_dir, "ml-10M100K")
+        return os.path.join(self.raw_dir, self.name)
+
+    @property
+    def graph_path(self):
+        return os.path.join(self.raw_path, self.name + ".bin")
+
+    @property
+    def feat_path(self):
+        return os.path.join(self.raw_path, self.name + "_feat.pkl")
+
+    @property
+    def info_path(self):
+        return os.path.join(self.raw_path, self.name + ".pkl")
+
+    def _process_user_fea(self, user_data):
+        if self.name == "ml-100k" or self.name == "ml-1m":
+            ages = user_data["age"].values.astype(np.float32)
+            gender = (user_data["gender"] == "F").values.astype(np.float32)
+            all_occupations = set(user_data["occupation"])
+            occupation_map = {ele: i for i, ele in enumerate(all_occupations)}
+            occupation_one_hot = np.zeros(
+                shape=(user_data.shape[0], len(all_occupations)),
+                dtype=np.float32,
+            )
+            occupation_one_hot[
+                np.arange(user_data.shape[0]),
+                np.array(
+                    [occupation_map[ele] for ele in user_data["occupation"]]
+                ),
+            ] = 1
+            user_features = np.concatenate(
+                [
+                    ages.reshape((user_data.shape[0], 1)) / 50.0,
+                    gender.reshape((user_data.shape[0], 1)),
+                    occupation_one_hot,
+                ],
+                axis=1,
+            )
+        elif self.name == "ml-10m":
+            user_features = np.zeros(
+                shape=(user_data.shape[0], 1), dtype=np.float32
+            )
+        else:
+            raise NotImplementedError
+        return user_features
+
+    def _process_movie_fea(self, movie_data):
+        import torchtext
+        from torchtext.data.utils import get_tokenizer
+
+        # Old torchtext-legacy API commented below
+        # TEXT = torchtext.legacy.data.Field(tokenize='spacy', tokenizer_language='en_core_web_sm')
+        tokenizer = get_tokenizer(
+            "spacy", language="en_core_web_sm"
+        )  # new API (torchtext 0.9+)
+        embedding = torchtext.vocab.GloVe(
+            name="840B", dim=300, cache=self.text_cache
+        )
+
```

Comment:
@mufeili There's one more issue that I suddenly found. torchtext tokenizer (spacy) requires users to download something in advance through `python -m spacy download en_core_web_sm`. I will check later what it is exactly. I think users would find it inconvenient.